### PR TITLE
Prepare CMake for Qt6

### PR DESCRIPTION
**Pull request by zneix**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Renamed variables of QT5KEYCHAIN to just QTKEYCHAIN due to how qt keychain library changes its target name from `qt5keychain` to `qt6keychain` if we build against Qt6.

⚠ Important note: qtkeychain already uses `BUILD_WITH_QT6` CMake parameter, so I thought it would be convenient to just reuse that. No idea if that's a bad practice to use one variable/option that affects multiple things at once, but pretty sure that in this scope it wouldn't change anything.

Required for Chatterino/libcommuni#2

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,11 +13,18 @@ option(BUILD_APP "Build Chatterino" ON)
...
+option(USE_SYSTEM_QTKEYCHAIN "Use system QtKeychain library" OFF)
 option(USE_PRECOMPILED_HEADERS "Use precompiled headers" ON)
+option(BUILD_WITH_QT6 "Use Qt6 instead of default Qt5" OFF)
 
 option(USE_CONAN "Use conan" OFF)
 
+if (BUILD_WITH_QT6)
+    set(MAJOR_QT_VERSION "6")
+else()
+    set(MAJOR_QT_VERSION "5")
```
Comment (zneix) [inline CMakeLists.txt:25]: Should this be strings or numbers? No idea how CMake treats things like this

Comment (AnotherFoxGuy) [reply CMakeLists.txt:25]: In cmake everything is a string, it doesn't have any other types

File: CMakeLists.txt [hunk 2]
```diff
@@ -13,11 +13,18 @@ option(BUILD_APP "Build Chatterino" ON)
 option(BUILD_TESTS "Build the tests for Chatterino" OFF)
 option(USE_SYSTEM_PAJLADA_SETTINGS "Use system pajlada settings library" OFF)
 option(USE_SYSTEM_LIBCOMMUNI "Use system communi library" OFF)
-option(USE_SYSTEM_QT5KEYCHAIN "Use system Qt5Keychain library" OFF)
+option(USE_SYSTEM_QTKEYCHAIN "Use system QtKeychain library" OFF)
```
Comment (pajlada) [inline CMakeLists.txt:16]: Make sure you add a dev changelog entry for this

Comment (zneix) [reply CMakeLists.txt:16]: Hopefully 7f8e05b is a good changelog entry for this one.

**Review by pajlada — APPROVED** [review]
:+1:

**Pull request merged**